lib/template-loader.php

```diff
@@ -383,3 +383,26 @@ function gutenberg_template_loader_filter_block_editor_settings( $settings ) {
 	return $settings;
 }
 add_filter( 'block_editor_settings', 'gutenberg_template_loader_filter_block_editor_settings' );
+
+/**
+ * Removes post details from block context when rendering a block template.
+ *
+ * @param array $context Default context.
+ *
+ * @return array Filtered context.
+ */
+function gutenberg_template_render_without_post_block_context( $context ) {
+	/*
+	 * Blocks rendered for a template should be treated as if there is no
+	 * current post, since the purpose of a template is in prescribing how
+	 * blocks appear for any post which matches the template. The post is
+	 * assigned only once the template matches in template loading.
+	 */
+	if ( isset( $context['postType'] ) && 'wp_template' === $context['postType'] ) {
```

Comment:
We need the same thing for template parts.

Reply:
> We need the same thing for template parts.

Thanks. Updated in rebased 1c5d9e4e010ecdefcde954f43a53b4458735ca7c.